src/frontend/qt_sdl/AudioSettingsDialog.cpp

```diff
@@ -68,6 +75,35 @@ AudioSettingsDialog::AudioSettingsDialog(QWidget* parent) : QDialog(parent), ui(
     ui->txtMicWavPath->setEnabled(iswav);
     ui->btnMicWavBrowse->setEnabled(iswav);
 
+    bool isext = (Config::MicInputType == 1);
+    ui->cbMic->setEnabled(isext);
+
+    #if QT_VERSION >= 0x060000
+        const auto deviceInfos = QMediaDevices::availableDevices(QAudioDevice::Input);
+        for (const QAudioDevice &deviceInfo : deviceInfos)
+        {
+            ui->cbMic->addItem(deviceInfo.description());
+        }
+        
+        if (Config::MicDevice == "")
+        {
+            Config::MicDevice = QAudioDevice::defaultInputDevice().deviceName().toStdString();
+        }
```

Comment:
This bit doesn't work in Qt 6 - see the macOS CI (which uses Qt 6 instead of Qt 5 like the rest).